### PR TITLE
Slo alerts

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Starting point to have CI gated on SLOs.

Follow up:
- Group existing tests on its own t.Run so alertSLOs can be TestAlertSLOs that run sequentially after all the others.
- Move the rules into its own consumable source so other entities can use them.
- Once stable gate presubmits on TestAlertSLOs failing.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -128,12 +136,133 @@ func main(m *testing.M) int {
...
 
 	// Everything's okay to run tests
 	log.Info("executing e2e tests", "options", globalOpts)
 	return m.Run()
 }
 
+// alertSLOs creates alert for our SLO/SLIs and log when firing.
+// TODO(alberto): have a global t.Run which runs all tests first then TestAlertSLOs.
+func alertSLOs() error {
+	slosRules := &monitoringv1.PrometheusRule{
```
Comment (csrwng) [inline test/e2e/e2e_test.go:149]: My preference would be to have these rules installed with hypershift (maybe optionally with a flag), in the same way that we install recording rules (https://github.com/openshift/hypershift/blob/main/cmd/install/assets/recordingrules/hypershift.yaml)
The yaml file imho may be an easier way to publish our SLOs for other folks to read/contribute (we can reference it from the docs)

Comment (enxebre) [reply test/e2e/e2e_test.go:149]: I agree, is the second bullet point in the PR desc.
But I can do it as part of this PR. Let me update.

**Comment by sjenning** [comment]
/lgtm
We will need to make sure we aren't going to flake tests before enforcing in CI

**Pull request merged**